sdk/servicebus/azure-servicebus/samples/sync_samples/session_pool_receive.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import uuid
+import concurrent
+
+from azure.servicebus import ServiceBusClient, Message, AutoLockRenew
+from azure.servicebus import NoActiveSession
+
+CONNECTION_STR = os.environ['SERVICE_BUS_CONNECTION_STR']
+# Note: This must be a session-enabled queue.
+QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
+
+def message_processing(sb_client, queue_name, messages):
+    while True:
+        try:
+            with sb_client.get_queue_session_receiver(queue_name, idle_timeout=1) as receiver:
+                renewer = AutoLockRenew()
+                renewer.register(receiver.session, timeout=None)
+                receiver.session.set_session_state("OPEN")
+                for message in receiver:
+                    messages.append(message)
+                    print("Message: {}".format(message))
+                    print("Time to live: {}".format(message.header.time_to_live))
+                    print("Sequence number: {}".format(message.sequence_number))
+                    print("Enqueue Sequence numger: {}".format(message.enqueue_sequence_number))
+                    print("Partition ID: {}".format(message.partition_id))
+                    print("Partition Key: {}".format(message.partition_key))
+                    print("Locked until: {}".format(message.locked_until_utc))
+                    print("Lock Token: {}".format(message.lock_token))
+                    print("Enqueued time: {}".format(message.enqueued_time_utc))
+                    message.complete()
+                    if str(message) == 'shutdown':
+                        receiver.session.set_session_state("CLOSED")
+                renewer.shutdown()
+        except NoActiveSession:
+            return
```

Comment:
We probably need to do something when "NoActiveSession" is encountered.
some useful logging information should be fine:
```
except NoActiveSession:
    log.info("Currently there is no available session in the entity, keep spinning waiting for next session"
    pass
```

Reply:
While that doesn't seem to address the issue the user is running into, it's a step better.  Will add.

This does lend a slight bit of credence to my "we should do it by explicit session IDs/use list_sessions" spitballing. (admittedly that has implications, but it would be one way around this).